sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -93,43 +93,53 @@ public TableAsyncClient buildAsyncClient() {
      * Sets the connection string to connect to the service.
      *
      * @param connectionString Connection string of the storage or CosmosDB table API account.
-     * @return The updated {@code TableClientBuilder}.
-     * @throws IllegalArgumentException if {@code connectionString} isn't a valid connection string.
+     *
+     * @return The updated {@link TableClientBuilder}.
+     *
+     * @throws IllegalArgumentException If {@code connectionString} isn't a valid connection string.
      */
     public TableClientBuilder connectionString(String connectionString) {
-        StorageConnectionString storageConnectionString
-            = StorageConnectionString.create(connectionString, logger);
+        StorageConnectionString storageConnectionString = StorageConnectionString.create(connectionString, logger);
         StorageEndpoint endpoint = storageConnectionString.getTableEndpoint();
+
         if (endpoint == null || endpoint.getPrimaryUri() == null) {
-            throw logger
-                .logExceptionAsError(new IllegalArgumentException(
-                    "connectionString missing required settings to derive tables service endpoint."));
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException(
+                    "'connectionString' missing required settings to derive tables service endpoint."));
```

Comment:
This should be split into NPE and IllegalArgumentException as well and also look at all other places where this needs to be updated.

Reply:
`connectionString` is used to obtain the `endpoint`, so it would only make sense to throw an NPE if `connectionString` itself is null, right? If `endpoint` ends up being `null` it would be because the connection string was wither malformed or invalid.

Reply:
Even so, the method `StorageConnectionString.create()` we use to parse a given `connectionString` will throw an `IllegalArgumentException` if it's `null`. Should we attempt to avoid that behavior and throw an NPE ourselves before calling it?